src/renderer/components/common/QrCode/QrReader/QrReader.tsx

```diff
@@ -128,9 +134,14 @@ const QrReader = ({
 
     if (fountainResult) {
       // decode the 1st frame --> it's a single frame QR
-      const result = EXPORT_ADDRESS.decode(fountainResult.slice(3));
+      let result: { payload: string | SeedInfo[] | { addr: SeedInfo } };
```

Comment:
`string | SeedInfo[] | { addr: SeedInfo } ` this bit repeats multiple times so its worth extracting it into a separate type

Reply:
Also, how `payload` here can possibly be a `string` given that we only assign it to `ExposreAddress.decode` or to `ExportAddress_2.decode`?

Reply:
It may be a `string` because the user may scan qr code with address. Like Nova Wallet does.

Reply:
`String` is possible because we're getting just a raw bytes from the camera device. We don't know what was scanned. It may be:
1. Raw string with address (like for Nova Wallet)
2. ExportAddrV1
3. ExportAddrV2
Depending on the decoding result the application will accordingly open the screen for:
1. Single Polkadot Vault pairing (legacy)
2. Multishard Polkadot Vault pairing (legacy)
3. Polkadot Vault pairing (with dynamic derivations)